MerlinAU.sh

```diff
@@ -499,37 +500,24 @@ _ScriptVersionStrToNum_()
    echo "$verNum" ; return 0
 }
 
-##---------------------------------------##
-## Added by ExtremeFiretop [2024-Jun-19] ##
-##---------------------------------------##
+##----------------------------------------##
+## Modified by Martinski W. [2024-Aug-11] ##
+##----------------------------------------##
 _GetFirmwareVariantFromRouter_()
 {
-   ##DEFAULTS TO MERLIN##
-   local retCode=0  newVersionStr
-
-   buildInfoStr="$(nvram get buildinfo)"
-   innerverStr="$(nvram get innerver)"
+   local hasGNUtonFW=false
 
    ##FOR TESTING/DEBUG ONLY##
-   if false # Change to true for forcing GNUton flag
-   then 
-      isGNUtonFW=true
-   else
-      # Check if innerver and fwInstalledExtendNum contains "gnuton"
-      if echo "$innerverStr" | grep -iq "gnuton" || \
-         echo "$fwInstalledExtendNum" | grep -iq "gnuton"
-      then
-          isGNUtonFW=true
-      # if the version string contain "merlin" 
-      elif echo "$buildInfoStr" | grep -iq "merlin"
-      then
-          isGNUtonFW=false
-      else
-          isGNUtonFW=false
-      fi
-   fi
+   if false  # Change to true for forcing GNUton flag #
+   then hasGNUtonFW=true ; return 0 ; fi
+   ##FOR TESTING/DEBUG ONLY##
+
+   # Check if installed F/W NVRAM vars contain "gnuton" #
+   if echo "$fwInstalledInnerVers" | grep -iq "gnuton" || \
+      echo "$fwInstalledExtendNum" | grep -iq "gnuton"
+   then hasGNUtonFW=true ; fi
```

Comment:
The function "`_GetFirmwareVariantFromRouter_()`" was modified to "clean up" the previously convoluted logic and have a simpler logic flow. Also, the code that was intended for "testing & debugging" purposes only should be clearly identified and separated from the rest of the code. This was not flagged by the Linter tool, but it was done to follow a coding standard & best practice.

Reply:
Thanks for the example!